experimenter/experimenter/klaatu/tests/test_client.py

```diff
@@ -0,0 +1,184 @@
+import datetime
+import unittest
+import zipfile
+from io import BytesIO
+from unittest import mock
+
+from experimenter.klaatu.client import (
+    KlaatuClient,
+    KlaatuError,
+    KlaatuStatus,
+    KlaatuTargets,
+    KlaatuWorkflows,
+)
+
+
+class TestKlaatuClient(unittest.TestCase):
+    def setUp(self):
+        self.client = KlaatuClient(KlaatuWorkflows.WINDOWS)
+
+    @mock.patch("experimenter.klaatu.client.requests.post")
+    def test_run_test_success(self, mock_post):
+        mock_post.return_value.status_code = 204
+
+        self.client.run_test(
+            experiment_slug="training-only-for-dev-tools",
+            branch_slug="control",
+            targets=[KlaatuTargets.LATEST_BETA, "137.0"],
+        )
+
+    @mock.patch("experimenter.klaatu.client.requests.post")
+    def test_run_test_raises_on_failed_post(self, mock_post):
+        mock_post.return_value.status_code = 400
+        mock_post.return_value.text = "Bad Request"
+
+        with self.assertRaises(KlaatuError) as error:
+            self.client.run_test("slug", "branch", [KlaatuTargets.LATEST_RELEASE])
+
+        self.assertIn("Failed to trigger workflow", str(error.exception))
```

Comment:
Oh funny, I would've thought the assert would need to be idented in the `with` block, but I guess its variable persists outside the block.